botocore/parsers.py

```diff
@@ -1173,7 +1178,7 @@ def _parse_non_payload_attrs(
         for name in member_shapes:
             member_shape = member_shapes[name]
             location = member_shape.serialization.get('location')
-            if location is None:
+            if location is None or location not in KNOWN_LOCATIONS:
```

Comment:
@ubaskota and I discussed offline. I agree that we don't need this line since the conditionals implicitly handle each know location.